controller/Jwks.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\controller;
+
+use oat\tao\helpers\UrlHelper;
+use oat\tao\model\security\Business\Contract\JwksRepositoryInterface;
+use oat\taoLti\models\classes\Platform\Service\CachedKeyChainGenerator;
+use oat\taoLti\models\classes\Security\DataAccess\Repository\CachedPlatformJwksRepository;
+use \tao_actions_CommonModule as CommonModule;
+use function GuzzleHttp\Psr7\stream_for;
+
+class Jwks extends CommonModule
+{
+    public function view(): void
+    {
+        $this->setData('jwks-key', json_encode($this->getJwksRepository()->find()));
+        $this->setData('jwks-generate-url', $this->getUrlGenerator()->buildUrl('index'));
+        $this->setView('jwks/Jwks.tpl');
+    }
+
+    public function index(): void
+    {
+        switch ($this->getRequestMethod()) {
+            case 'POST':
+                $this->postJwks();
+                break;
+
+            default:
+                $this->setResponse(
+                    $this->getPsrResponse()
+                        ->withStatus(501)
+                        ->withBody(stream_for(__('Not Implemented')))
+                );
+        }
```

Comment:
```suggestion
       try {
           $this->getCachedKeyChainGenerator()->generate();
        
           $this->setSuccessJsonResponse([]);
        
           if ($this->getRequestMethod() !== 'POST') {
               Throw new common_exception_MethodNotAllowed('Not Implemented')
           }
       } catch (common_exception_MethodNotAllowed $exception) {
           $this->setErrorJsonResponse($exception->getMessage(), 0, [], 501);
       } catch (Throwable $exception) {
           $this->setErrorJsonResponse($exception->getMessage(), 0, [], 501);
       }
```
- I think we should use our standard to the response. 

- Switch case encourages developers to increase the complexity of this method. We should avoid it, on OO we have polymorphism to avoid it. The class method should be POST strictly.

- I would not make the `index` endpoint deal with all cases. If we have to put `index` in the URL it is a limitation of our routing.

Reply:
What should I privilege, RestApi or OpenClosed principle? As we are not handling HTTP method in our router, we have to choose. If we are using, one class per Method, we cannot have `REST Resource Naming Best Practices`

Reply:
We should privilege both, but we have naming limitations and I am not arguing with that. 
Is it not recommended to handle 2 verbs in on method and adding the switch case you are doing that. May comment was also about use the standard response. The call will be done by FE and they expect something like this.